test_unstructured/partition/common/test_metadata.py

```diff
@@ -62,73 +62,139 @@ def but_it_returns_None_when_there_is_no_file_at_that_path(self, tmp_path: pathl
-    assert elements[10].metadata.parent_id == elements[8].id, "ListItem should be child of Title 2"
-    assert elements[11].metadata.parent_id == elements[8].id, "Text should be child of Title 2"
-
-
-def test_set_element_hierarchy_custom_rule_set():
-    elements_to_set = [
-        Header(text="Header"),  # 0
-        Title(text="Title"),  # 1
-        NarrativeText(text="NarrativeText"),  # 2
-        Text(text="Text"),  # 3
-        Title(text="Title 2"),  # 4
-        FigureCaption(text="FigureCaption"),  # 5
-    ]
+class Describe_set_element_hierarchy:
+
+    def it_applies_default_ruleset(self):
+        elements = [
+            Title(element_id="0", text="Title0"),
+            Text(element_id="1", text="Text0"),
+            Header(element_id="2", text="Header0"),
+            Text(element_id="3", text="Text1"),
+            Title(element_id="4", text="Title1"),
+            Text(element_id="5", text="Text2"),
+        ]
+
+        result = set_element_hierarchy(elements)
+
+        assert result[0].metadata.parent_id is None
+        assert result[1].metadata.parent_id == "0"  # Text0 is under Title0
+        assert result[2].metadata.parent_id is None  # Header0 is higher than Title0
+        assert result[3].metadata.parent_id == "2"  # Text1 is under Header0
+        assert result[4].metadata.parent_id == "2"  # Title1 is under Header0
+        assert result[5].metadata.parent_id == "4"  # Text2 is under Title1, which is under Header0
+
+    def it_applies_ruleset_based_on_element_category_regardless_of_depth(self):
+        elements = [
+            Title(element_id="0", text="Title", metadata=ElementMetadata(category_depth=2)),
+            Text(element_id="1", text="Text", metadata=ElementMetadata(category_depth=1)),
+            Header(element_id="2", text="Header", metadata=ElementMetadata(category_depth=2)),
+            Text(element_id="3", text="Text", metadata=ElementMetadata(category_depth=1)),
+        ]
+
+        result = set_element_hierarchy(elements)
+
+        assert result[1].metadata.parent_id == "0"  # Text is under Title despite category_depth
+        assert result[3].metadata.parent_id == "2"  # Text is under Header despite category_depth
+
+    def but_it_applies_category_depth_when_element_category_is_the_same(self):
+        elements = [
+            Title(element_id="0", text="Title0", metadata=ElementMetadata(category_depth=1)),
+            Text(element_id="1", text="Text0", metadata=ElementMetadata(category_depth=0)),
+            Text(element_id="2", text="Text1", metadata=ElementMetadata(category_depth=1)),
+            Text(element_id="3", text="Text2", metadata=ElementMetadata(category_depth=0)),
+        ]
+
+        result = set_element_hierarchy(elements)
+
+        assert result[0].metadata.parent_id is None
+        assert result[1].metadata.parent_id == "0"
+        assert result[2].metadata.parent_id == "1"
+        assert result[3].metadata.parent_id == "0"
```

Comment:
Hmm, this behavior seems a little odd. I'm wondering when we would even use this sort of thing.

It strikes me that not just any old element should be able to be a parent. Like why would we make one `Text` element a child of another?

Strikes me this would be the distinction between nodes and leaves, maybe container-nodes and leaf-nodes in the hierarchy. `Title` can definitely be a parent because a document-section "contains" paragraphs. But I would expect `NarrativeText` for example to be a leaf-node.

More of a discussion topic I suppose :)

Reply:
I think this applies mostly (or only) to `ListItem`s, but I just used `Text` for copy/paste reasons. I can change it to `ListItem` if you think that'd be better.

Reply:
I would. It's helpful to the reader to see typical usage as much as possible I'd say. One of the benefits of reading the tests should be seeing examples of how one might want to use the units under test.